Sources/CreateAPI/Generator/Generator+Schemas.swift

```diff
@@ -687,10 +687,42 @@ extension Generator {
             if context.isPatch && isOptional && options.paths.makeOptionalPatchParametersDoubleOptional {
                 type = type.asPatchParameter()
             }
+
             var defaultValue: String?
             if options.entities.includeDefaultValues {
-                if type.isBool {
-                    defaultValue = (info?.defaultValue?.value as? Bool).map { $0 ? "true" : "false" }
+                func describeDefaultValue(_ anyValue: AnyCodable) -> String? {
+                    if let bool = anyValue.value as? Bool {
+                        return bool ? "true" : "false"
+                    } else if let int = anyValue.value as? Int {
+                        return "\(int)"
+                    } else if let uint = anyValue.value as? UInt {
+                        return "\(uint)"
+                    } else if let double = anyValue.value as? Double {
+                        return "\(double)"
+                    } else if let string = anyValue.value as? String {
+                        if let formatString = info?.formatString, !formatString.isEmpty {
+                            return nil
+                        }
+                        // Surround the value in quotes, otherwise the string is rendered improperly.
+                        return "\"\(string)\""
+                    } else if let array = anyValue.value as? [AnyCodable] {
+                        return "\(array.map(describeDefaultValue(_:)))"
+                    } else if let dictionary = anyValue.value as? [String: AnyCodable] {
+                        return "\(dictionary.mapValues(describeDefaultValue(_:)))"
+                    } else {
+                        return nil
+                    }
```

Comment:
Nice approach, but I think there are a few scenarios where this doesn't hold up. 

Currently we assume that if there is a `Int`/`Uint`/`Double`/`String`/`Array`/`Dictionary` default value then the property is also that same type, but that is not guaranteed to always be the case.

I see that you accounted for this by checking `formatString`, but additionally a user can override the [`dataTypes`](https://github.com/CreateAPI/CreateAPI/blob/main/Docs/ConfigOptions.md#datatypes) configuration option to switch out types based on formats as they see fit. For example, while a `string` with the `uri` format will default to the `URL` type, the user could disable that by overriding the `dataType` and setting it back to `String`. 

I think a more robust way around this might be to compare the type of the property and then only try to unwrap the default value as the corresponding type if the property can be expressed as a boolean, integer, float or string literals? 

While we can't check for `ExpressibleByXX` conformance directly, we can hardcode a set of types that we know about (`String`, `Double`, `Float`, `Int`, `Int32` etc), so I'm wondering if we can do something like this:

```swift
if type.isExpressibleByStringLiteral, let defaultValue = info?.defaultValue.stringLiteral {
    defaultValue = defaultValue
} else if type.isExpressibleByIntegerLiteral, let defaultValue = ... {
    defaultValue = defaultValue
} else if type.isEnum ... {
    defaultValue = ".\(...)"
}
```

As for Arrays and Dictionaries, I think they might be more complicated than they seem... An array literal seems doable, providing the item schema meets one of the examples above, but admittedly it might not be necessary just yet? As for the dictionary literal, that would only work if the property wasn't a schema object and was instead a literal dictionary literal. I think it's perfectly correct for a property of a json object schema to have a `defaultValue` that is a json object dictionary which would end up generating code that doesn't compile.

---

I haven't dug deep into how you'd go about doing this, but `TypeIdentifier` seems to have some properties already (like `isString`) that will help. Let me know if you have any questions!

Reply:
Cool, I'll tinker some more with this.